includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -380,8 +380,7 @@ public function get_product_data() {
 			];
 		}
 
-		$data['displayItems'] = $items;
-		$data['total']        = [
+		$data['total'] = [
```

Comment:
Yes you are correct here. Unfortunately, as I mentioned `total` is required on the product page to initialize the PR button. It will never be used though since later it will be replaced with the response of `add_to_cart`.

If you want I can change it to I can return a dummy object like: 

```php
$data['total'] = ['label' => 'anything', 'amount' => 0, 'pending' => true],
```

It will work but I think the real fix is to address my [comment](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1577/files#r645288392).

Reply:
I don't think a total of 0 is allowed. You can use a [common minimum amount](https://stripe.com/docs/currencies#minimum-and-maximum-charge-amounts) for a test payment as I mentioned we do in the blocks implementation, and then we can discard all this back-end code.

Reply:
I'm going to create an issue later on to discuss how we can consolidate how the PR button is initialized. It will fix comments related to  `get_product_data`.

Reply:
> I think we can at least remove $items from here as it's not being used anymore.

done 3caec51